geostore/check_stac_metadata/utils.py

```diff
@@ -294,6 +313,33 @@ def report_duplicate_object_names(object_pairs: List[Tuple[str, Any]]) -> JsonOb
 
         return report_duplicate_object_names
 
+    def check_if_contains_assets(self, metadata_url: str) -> None:
```

Comment:
I prefer to export this logic into a separate function to be easier to read and test.